paddle/fluid/framework/fleet/heter_wrapper.cc

```diff
@@ -116,14 +116,12 @@ void HeterWrapper::SerializeToReq(const std::string& varname, Scope* scope,
            tensor->numel() * SizeOfType(tensor->type()));
   } else {
 #if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
-    memory::Copy(platform::CPUPlace(), data_ptr,
-                 BOOST_GET_CONST(platform::CUDAPlace, tensor->place()),
+    memory::Copy(platform::CPUPlace(), data_ptr, tensor->place(),
```

Comment:
// TODO(Aurelius84): In next PR, we will consider implement `PLACE_GET` macro  here to help Template instantiation.